docs/data/material/components/grid2/grid2.md

```diff
@@ -0,0 +1,245 @@
+
+```js
+import { ThemeProvider, createTheme } from '@mui/material/styles';
+
+function Demo() {
+  return (
+    <ThemeProvider
+      theme={createTheme({
+        breakpoints: {
+          values: {
+            laptop: 1024,
+            tablet: 640,
+            mobile: 0,
+            desktop: 1280,
+          },
+        },
+      })}
+    >
+      <Grid container spacing={{ mobile: 1, tablet: 2, laptop: 3 }}>
+        {Array.from(Array(4)).map((_, index) => (
+          <Grid mobile={6} tablet={4} laptop={3} key={index}>
+            <div>{index + 1}</div>
+          </Grid>
+        ))}
+      </Grid>
+    </ThemeProvider>
+  );
+}
+```
+
+:::info
+The custom breakpoints affect both the size and offset props:
+
+```diff
+- <Grid xs={6} xsOffset={2}>
++ <Grid mobile={6} mobileOffset={2}>
+```
+
+:::
+
+### TypeScript
+
+You have to set module augmentation on the theme breakpoints interface. The properties with `true` value will appear as `{key}`(size prop) and `{key}Offset`(offset prop).
+
+```ts
+declare module '@mui/system' {
+  interface BreakpointOverrides {
+    // Your custom breakpoints
+    laptop: true;
+    tablet: true;
+    mobile: true;
+    desktop: true;
+    // Remove default breakpoints
+    xs: false;
+    sm: false;
+    md: false;
+    lg: false;
+    xl: false;
+  }
+}
+```
+
+## Prevent scrollbar
+
+If you use grid as a container in a small viewport, you might see a horizontal scrollbar because the negative margin is applied on all sides of the grid container.
+
+To prevent the scrollbar, set `disableEqualOverflow` prop to `true`. It will enable negative margin only on the top and left sides of the grid which remove overflow on the right-hand side.
+
+{{"demo": "OverflowGrid.js", "bg": true}}
+
+:::warning
+You should avoid adding borders or background to the grid when `disableEqualOverflow: true` because the negative margin (applied only at the top and left sides) makes the grid visually misaligned.
+:::
+
+## Customization
+
+### Centered elements
+
+If you want to make the content of the grid item centered, specify `display="flex"` directly to the grid item. Then, use `justifyContent` and/or `alignItems` to adjust the position of the content.
+
+{{"demo": "CenteredElementGrid.js", "bg": true}}
+
+:::warning
+Using `container` prop would not work because the grid container is designed to wrap only the grid items, not random elements.
+:::
+
+### Full bordered
+
+{{"demo": "FullBorderedGrid.js"}}
+
+### Half bordered
+
+{{"demo": "HalfBorderedGrid.js"}}
+
+## Limitations
+
+### direction column and column-reverse
+
+The column width (`xs`, ..., `xl`) and offset props are **not supported** within `direction="column"` and `direction="column-reverse"` containers.
+
+They define the number of grids the component will use for a given breakpoint. They are intended to control **width** using `flex-basis` in `row` containers but they will impact height in `column` containers.
```

Comment:
"They" is ambiguous here. Does it refer to the size & offset props, or the direction=x properties?

Reply:
it refers to the size & offset props.

Reply:
How about this? 

```suggestion
Size and offset props define the number of columns the component will use for a given breakpoint. 
They are intended to control the width using `flex-basis` in `row` containers, but they will impact the height in `column` containers.
```